app/controllers/api/v1/users_controller.rb

```diff
@@ -6,9 +6,34 @@ class UsersController < Api::V1::ApiController
       # PUT /api/v1/users/:id
       def update
         if current_user.update(user_params)
-          render json: current_user
+          render json: {
+            user_id: current_user.id, 
+            name: current_user.name, 
+            email: current_user.email,
+            authentication_token: current_user.authentication_token
+          }
         else
-          render json: { error: current_user.errors.as_json }, status: :bad_request
+          render json: { errors: current_user.errors.as_json }, status: :bad_request
+        end
+      end
+
+      def reset_password
+        render json: { errors: 'The password is incorrect'}, status: :bad_request and return unless 
```

Comment:
This is more readable if the conditions is before the code to be executed, like this:

```
unless current_user.valid_password?(user_password[:prev])
    render json: { errors: 'The password is incorrect'}, status: :bad_request and return
end
```